CONTRIBUTING.md

```diff
@@ -109,6 +106,66 @@ The contribution process is outlined below:
    maintainer. At this point, a Meta employee will be notified to merge your PR,
    due to tooling limitations.
 
+## Commit Messages
+
+We build Velox for the long-run, and to do so it is crucial that project
+maintainers are able to efficiently inspect and understand project logs.
+
+Commit messages that follow a strict pattern improve maintainability, and allow
+tasks such as summarizing changelogs and identifying API breaking changes to be
+automated. Despite requiring more rigor from authors and reviewers, consistency
+creates productivity in the long term.
+
+In Velox, commit messages are generated based on the input provided to PRs, and
+must follow the [conventional commit
+specification](https://www.conventionalcommits.org/en/v1.0.0/) in the following
+manner:
+
+**PR titles** must follow the pattern:
+
+> \<type\>[optional scope]: \<description\>
+
+where:
+
+* *Type* can be any of the following keywords:
+  * **feat** when new features are being added.
+  * **fix** for bug fixes.
+  * **build** for build or CI-related improvements.
+  * **docs** for enhancements to documentation (only).
+  * **refactor** for refactoring (no logic changes).
+
+* PR titles also take an *optional scope* field containing the area
+  of the code being target by the PR, to further help commit classification.
+  For example, "fix(expression): " or "refactor(parquet): " or "feat(memory):".
+  It is ok to omit this field if there is no adequate classification for the
+  PR, or if the PR touches multiple different components.
+
+* A *description* sentence summarizing the PR, written in imperative tone. The
+  description must be capitalized, not contain a period at the end, and be kept
+  under a 80 characters limit.
+
+Examples of PR titles are:
+
+* feature(type): Add support for IPPREFIX type
+* fix: Prevent unnecessary flatmap to map conversion
+* refactor(vector): Use 'if constexpr' for Buffer::is_pod_like_v\<T\>
+
+The **PR body** must contain a summary of the change, focusing on the *what*
+and *why*. Keep columns at 80 characters.
+
+**Breaking API Changes.** A "BREAKING CHANGE:" footer must be added to PRs
+that break backwards compatibility of any external API in Velox, followed by a
+sentence explaining the extent of the API change. This means either API changes
+that may break client builds, or semantic changes on the behavior of such APIs.
+
+If there is a Github Issue or Discussion related to the PR, authors must also add a
+"Fixes: <link>" section to the bottom of the PR body.
```

Comment:
There are multiple syntaxes that work. I use Fixes #<issue number>

I don't have a link, but I'm sure GitHub docs list all supported combinations somewhere. The point of doing this is to allow GitHub auto-close issues when PRs are merged. That said, one needs to be careful to not use this for PRs that partially fix an issue. For these I use "Part of #<issue number>" to allow GitHub to link PR to issue, but not auto-close the issue.

Reply:
Found the docs here:

> https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue

Looks like there are multiple keywords that work (including "fixes"), but "part of" is not listed in any of the docs. I suppose this is because github ends up linking the PR to the Issue since the Issue was mentioned (but does not automatically close the associated Issue).

But +1 on adding guidelines for "Fixes" and "Part of", depending on what authors want the semantic to be; will update the file.